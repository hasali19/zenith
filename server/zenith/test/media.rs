use insta::assert_json_snapshot;
use test_macros::test;

use crate::{with_app, TestApp};

macro_rules! test_snapshot {
    ($name:ident, $path:expr) => {
        #[test(with_app)]
        async fn $name(app: TestApp) {
            assert_json_snapshot!(app.get($path).await);
        }
    };
}

test_snapshot!(get_all_movies, "/movies");
test_snapshot!(get_movie_by_id, "/movies/1");
test_snapshot!(get_all_tv_shows, "/tv/shows");
test_snapshot!(get_show_by_id, "/tv/shows/4");
test_snapshot!(get_seasons_for_show, "/tv/shows/4/seasons");
test_snapshot!(get_season_by_id, "/tv/seasons/7");
test_snapshot!(get_episodes_for_show, "/tv/shows/4/episodes");
test_snapshot!(get_episodes_for_season, "/tv/seasons/7/episodes");
test_snapshot!(get_episode_by_id, "/tv/episodes/9");
