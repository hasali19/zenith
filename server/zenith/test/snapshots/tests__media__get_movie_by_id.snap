---
source: server/zenith/test/media.rs
assertion_line: 16
expression: "app.get(\"/movies/1\").await"
---
{
  "backdrop": null,
  "external_ids": {
    "tmdb": null
  },
  "id": 1,
  "overview": null,
  "poster": null,
  "release_date": null,
  "title": "Test Movie 1",
  "user_data": {
    "is_watched": false,
    "last_watched_at": null,
    "position": null
  },
  "video_info": {
    "audio": [],
    "duration": 0.0,
    "format": null,
    "path": "/path/to/Test Movie 1/Test Movie 1.mp4",
    "subtitles": [],
    "video": null
  }
}
